### PR TITLE
UI: Add a widget and some notifications for external linter

### PR DESCRIPTION
**Pull request by mchernyavsky**
Screenshot 2021-11-12 at 02 18 21

Screenshot 2021-11-12 at 04 02 08

Screenshot 2021-11-12 at 02 20 45 Screenshot 2021-11-12 at 02 20 26

Also, see `org.rust.external.linter.max.duration` registry option.

changelog: Add a widget for the external linter and notify a user about the linter is turned on/off or if the linter significantly affects performance on the project

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/status/RsExternalLinterWidget.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        project.messageBus.connect(this).subscribe(RustProjectSettingsService.RUST_SETTINGS_TOPIC, object : RustSettingsListener {
+            override fun rustSettingsChanged(e: RustSettingsChangedEvent) {
+                if (e.isChanged(State::externalLinter) || e.isChanged(State::runExternalLinterOnTheFly)) {
+                    update()
+                }
+            }
+        })
+
+        project.service<RsExternalLinterTooltipService>().showTooltip(this)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/status/RsExternalLinterWidget.kt:69]: I think the popup is shown too early

**Review by vlad20012 — APPROVED** [review]
bors r+

**Comment by vlad20012** [comment]
This should be documented on this page https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-analysis.html#external-linters

**Comment by MarinaKalashina** [comment]
Added to the docs https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-analysis.html#external-linters

**Pull request merged**